terraform/jenkins/jenkins_controller.tf

```diff
@@ -0,0 +1,102 @@
+# SPDX-FileCopyrightText: 2023 Technology Innovation Institute (TII)
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# Build the Jenkins controller image
+module "jenkins_controller_image" {
+  source = "../../tf-modules/azurerm-nix-vm-image"
+
+  nix_attrpath   = "outputs.nixosConfigurations.jenkins-controller.config.system.build.azureImage"
+  nix_entrypoint = "${path.module}/../.."
+
+
+  name                = "jenkins-controller"
+  resource_group_name = azurerm_resource_group.default.name
+  location            = azurerm_resource_group.default.location
+
+  storage_account_name   = azurerm_storage_account.vm_images.name
+  storage_container_name = azurerm_storage_container.vm_images.name
+}
+
+# Create a machine using this image
+module "jenkins_controller_vm" {
+  source = "../../tf-modules/azurerm-linux-vm"
+
+  resource_group_name = azurerm_resource_group.default.name
+  location            = azurerm_resource_group.default.location
+
+  virtual_machine_name         = "ghaf-jenkins-controller"
+  virtual_machine_size         = "Standard_D1_v2"
+  virtual_machine_source_image = module.jenkins_controller_image.image_id
+
+  virtual_machine_custom_data = join("\n", ["#cloud-config", yamlencode({
+    users = [
+      for user in toset(["bmg", "flokli", "hrosten"]) : {
+        name                = user
+        sudo                = "ALL=(ALL) NOPASSWD:ALL"
```

Comment:
Debatable, but I would rather disable SSH password auth (keep it only key based) and require passwords for sudo. That would require setting up some initial password to let user define proper password on first login, but will limit the security risks in case of compromised account.

Reply:
This is the current behaviour, wheel doesn't need password: https://github.com/search?q=repo%3Atiiuae%2Fghaf-infra%20wheelneedspassword&type=code

These machines are cattle, all user state and interactively-chosen passwords are wiped whenever the machine is recreated (when we boot a new image or change the user data).

We could potentially track a per user password hash here, and require that password to be entered for sudo, but it's a lot of churn.

Let's discuss this in a followup. We mirror the current behaviour, and I'd rather solve this problem more elegantly with something like SSO for ssh, where we can require an arbitrary set of MFA at the identity provider.

Reply:
Yeah, I agree we have made some mistakes in the past and I understand the convenience vs security debate, which is much older than myself. :) 
Yes, let's take it up on a followup. 
(SSO sounds a bit like an overkill for this, but let's see).